src/software/ai/hl/stp/stp.cpp

```diff
@@ -137,17 +139,44 @@ std::vector<std::shared_ptr<Tactic>> STP::assignRobotsToTactics(
     // https://github.com/saebyn/munkres-cpp is the implementation of the Hungarian
     // algorithm that we use here
 
-    if (world.friendlyTeam().numRobots() < tactics.size())
+    auto friendly_team         = world.friendlyTeam();
+    auto& friendly_team_robots = friendly_team.getAllRobots();
+
+    // Special handling for the Goalie tactics, since only one robot per team is permitted
+    // to act as the goalie
+    const std::optional<Robot> goalie    = friendly_team.goalie();
+    std::vector<Robot> non_goalie_robots = friendly_team_robots;
+    auto isGoalieTactic                  = [](std::shared_ptr<Tactic> tactic) {
+        return tactic->isGoalieTactic();
+    };
+
+    auto it = tactics.begin();
+    if (goalie)
+    {
+        non_goalie_robots.erase(
+            std::find(non_goalie_robots.begin(), non_goalie_robots.end(), *goalie));
+
+        // Assign the goalie to the first goalie tactic
+        auto iter = std::find_if(tactics.begin(), tactics.end(), isGoalieTactic);
+        if (iter != tactics.end())
+        {
+            (*iter)->updateRobot(*goalie);
+        }
+    }
+
+    // Discard all goalie tactics, since we have already assigned the goalie robot (if
+    // there is one) to the first goalie tactic, and there should only ever be one goalie
+    tactics.erase(std::remove_if(tactics.begin(), tactics.end(), isGoalieTactic),
+                  tactics.end());
+
+    if (non_goalie_robots.size() < tactics.size())
     {
         // We do not have enough robots to assign all the tactics to. We "drop"
         // (aka don't assign) the tactics at the end of the vector since they are
         // considered lower priority
         tactics.resize(world.friendlyTeam().numRobots());
```

Comment:
Does this not include 1 additional Tactic?
The goalie tactics are removed, but friendlyTeam().numRobots() should return the team size including the goalie?

Reply:
Great catch! changed to `        tactics.resize(non_goalie_robots.size());`

Reply:
I don't see the change?

Reply:
Could you double check?
![Screenshot from 2019-12-29 13-34-00](https://user-images.githubusercontent.com/9075711/71560400-e5818b80-2a3f-11ea-98e2-2ad0c37789e5.png)

Reply:
I see it now :+1: